### PR TITLE
chore: Remove main entity from user

### PR DESCRIPTION
**Pull request by sradevski**
This one has a lot more cleanup in it than the removal of the main entity

### CODE REVIEW

File: packages/modules/user/src/services/user-module.ts [hunk 1]
```diff
@@ -62,21 +62,42 @@ export default class UserModuleService<
...
   ): Promise<UserTypes.InviteDTO> {
-    const invite = await this.inviteService_.validateInviteToken(
-      token,
+    const jwtSecret = this.moduleDeclaration["jwt_secret"]
+    if (!jwtSecret) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        "No jwt_secret was provided in the UserModule's options. Please add one."
+      )
+    }
```
Comment (olivermrbl) [inline packages/modules/user/src/services/user-module.ts]: thought: We throw this error in two places in critical flows. Would it make sense to check for this in the service constructor and throw at registration-time instead?

So when set on `this.config`

**Review by olivermrbl — APPROVED** [review]
Overall, LGTM

File: packages/modules/user/src/services/user-module.ts (hunk 1 above)
Comment (adrien2p) [reply packages/modules/user/src/services/user-module.ts]: Yes I think it would be better to not get the error too later

**Review by adrien2p — APPROVED** [review]

File: packages/modules/user/src/services/user-module.ts (hunk 1 above)
Comment (sradevski) [reply packages/modules/user/src/services/user-module.ts]: Yes definitely, let me clean it up.

**Pull request merged**